pkg/binpack/single_az_pack_tightly_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binpack
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/palantir/k8s-spark-scheduler-lib/pkg/resources"
+)
+
+func TestSingleAZTightlyPack(t *testing.T) {
+	tests := []struct {
+		name                    string
+		driverResources         *resources.Resources
+		executorResources       *resources.Resources
+		numExecutors            int
+		nodesSchedulingMetadata resources.NodeGroupSchedulingMetadata
+		nodePriorityOrder       []string
+		expectedDriverNode      string
+		willFit                 bool
+		expectedCounts          map[string]int
+	}{{
+		name:              "picks the first zone when application fits entirely in either of the zones",
```

Comment:
Why is this something we want? I would argue that in a tie, it should pick nodes which are better for other preferences (such as the node age as the docs on the `SingleAZTightlyPack` suggest) or distribute evenly across zones

Reply:
This was the behaviour in AzAwareTightlyPack, I was trying to mimic the [exact behaviour](https://github.com/palantir/k8s-spark-scheduler-lib/blob/master/pkg/binpack/az_aware_pack_tightly_test.go#L37-L46) of that binpacking algo exactly apart from scheduling across AZ boundaries so we can have a direct comparison. Happy to change this in future, but I do think this should be in a subsequent PR so that we can perform a better comparison right now, I can add an issue for this

Reply:
https://github.com/palantir/k8s-spark-scheduler-lib/issues/74

Reply:
That makes sense, thanks for the clarification.
It is worth checking (in a follow up issue) both algorithms for this then as the presence of the explicit test indicates to me that this is a desired feature, but I can tell why it would be better than the other options and not worse.